superset-frontend/src/addSlice/AddSliceContainer.tsx

```diff
@@ -256,15 +264,12 @@ export default class AddSliceContainer extends React.PureComponent<
       const list: {
         label: string;
         value: string;
-      }[] = response.json.result
-        .map((item: Dataset) => ({
-          value: `${item.id}__${item.datasource_type}`,
-          label: this.newLabel(item),
-          labelText: item.table_name,
-        }))
-        .sort((a: { labelText: string }, b: { labelText: string }) =>
```

Comment:
fun fact, this didn't do anything since the selector re-sorts the options again :P Adding a `filterSort` prop to the selector is how you actually sort the items in the dropdown

Reply:
@etr2460 `filterSort` only sorts when searching. If the user just opens the select the values will be unsorted if we remove these lines. We'll probably need to apply your sort function here as well 😉 

https://user-images.githubusercontent.com/70410625/130602788-53badf35-a202-4910-840f-2ce71746dabf.mov

Reply:
@etr2460 The ideal solution would be to disable the `filterSort` when the Select is in async mode and sort the values on the server-side to correctly handle the pagination. This endpoint is not respecting the `order_column` and `order_direction` props, so we would need to fix that.

Reply:
hmm, interesting, i didn't even see results rendering when clicking into the modal without typing anything. although maybe that's because in our env we have too many datasets so it didn't render fast enough. i'll see what i can do to fix the sorting for all use cases here

Reply:
@etr2460 for the scenario where the paginated query with no search (100 results by default) is heavy, we created a prop called `fetchOnlyOnSearch`. What we need to do is make this configurable by organizations.